### ISSUE TITLE
Dysfunctional admonition \py@lightbox in sphinx.sty (latex converter)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2172]
The command `\py@lightbox` for admonitions of the `note`, `hint`, `important`, `tip` types is defined in `sphinx.sty` for latex as follows:

``` latex
\newcommand{\py@lightbox}{{%
  \setlength\parskip{0pt}\par
  \noindent\rule[0ex]{\linewidth}{0.5pt}%
  \par\noindent\vspace{-0.5ex}%
  }}
```

But the `\vspace` in the last line should have been before the `\noindent`. As it stands this will insert a backward vertical shift between the first and second line of the admonition text, resulting in poor pdf output. (as I have experienced in my current project).

I am adding to my preamble: 

``` latex
\renewcommand{\py@lightbox}{{\parskip\z@skip
  \par\allowbreak
  \noindent\rule{\linewidth}{0.5pt}\par\nobreak
  \noindent
  }}
\renewcommand{\py@endlightbox}{{\parskip\z@skip
  \par\nobreak
  \noindent\rule[.4\baselineskip]{\linewidth}{0.5pt}\par
  }}
```

where also the `\py@endlightbox` is redefined. This gives satisfying results in my use case but should be tested by others. The macros are designed not to perturb (if parskip is zero) the grid like spacing of lines. 

For example for two column output with no parskip, if an admonition appears on the left, the macros are designed in order for the text lines on the left to remain aligned with the ones on the right.

Interested people should test to see what I mean.

**Comment by tk0miya** [comment]
I feel your new `\py@lightbox` seems good.
But I don't reproduce the bad case.

Could you tell me about your _poor pdf output_?
(any examples or screnshots are very helpful for me)

**Comment by jfbu** [comment]
@tk0miya Hi, I create a new empty project named `TestLightbox` with the script `sphinx-quickstart` (currently on my system v1.3.1 from an Anaconda install), accepting all default answers (except `Create Windows command file?` as I am on Mac OS X). Then I modify `intro.rst` to contain in particular:

```
    Contents:

.. toctree::
   :maxdepth: 2

   testlightbox
```

and the `testlightbox.rst` file contents are:

```
===========
 Main file
===========


.. note::

   Sphinx is a tool that makes it easy to create intelligent and beautiful
   documentation, written by Georg Brandl and licensed under the BSD license.
   It was originally created for the new Python documentation, and it has
   excellent facilities for the documentation of Python projects, but C/C++ is
   already supported as well, and it is planned to add special support for other
   languages as well. Of course, this site is also created from reStructuredText
   sources using Sphinx! 
```

I then do `make latexpdf` and here is an image of the relevant page in the output:

![sphinxlatexlightbox](https://cloud.githubusercontent.com/assets/2589111/12175655/be06d042-b563-11e5-86f3-ccddc193d46b.png)

You can see the problem that the second line of the paragraph is too close to the first line.

I am attaching `index.rst` and `testlightbox.rst` to this comment with `.txt` extension.
[testlightbox.rst.txt](https://github.com/sphinx-doc/sphinx/files/81387/testlightbox.rst.txt)
[index.rst.txt](https://github.com/sphinx-doc/sphinx/files/81389/index.rst.txt)

Manually adding to the `.tex` file just before `\begin{document}` the proposed definitions of `\py@lightbox` and `\py@endlightbox` (within  `\makeatletter/\makeatother`), and then running `pdflatex`,  the line spacing is good.

**Comment by tk0miya** [comment]
Thank you. Exactly, it looks bad.
I'll try your macros and introduce to Sphinx later (maybe in this weekend).

**Comment by jfbu** [comment]
My macros do not try to maintain the same vertical layout as the original macros, because they were for my own project. It is a bit cumbersome to try to fix the "2nd line too close to 1st line" problem and at the same time make it so that documents will have the same global vertical layout, hence exactly the same page breaks as with current code in `sphinx.sty` ...

I have re-examined my macros with regards to `\parskip` (my project had nothing special for `\parskip` hence I did not bother initially handling it seriously).

Here is a better version:

``` latex
\newcommand{\py@lightbox}{%
  \par\allowbreak
  \noindent\rule{\linewidth}{0.5pt}\par\nobreak 
  {\parskip\z@skip\noindent}%
  }
\newcommand{\py@endlightbox}{%
  \par\nobreak
  {\parskip\z@skip\noindent\rule[.4\baselineskip]{\linewidth}{0.5pt}}\par
  }
```

Explanation: the `\setlength{\parskip}{0pt}` of the original code clearly is there for the added horizontal rules delimiting the "box" to be put at controlled vertical positions. `LaTeX` looks at the value of `\parskip` when it _starts_ a paragraph, for example due to `\noindent` or due to the first letter of a word. There is no reason in case `\parskip` has some positive value that the initial horizontal rule starting the annotation is closer to the previous paragraph than a normal inter-paragraph distance: hence we conclude that the `\setlength{\parskip}{0pt}`  of the original code was issued too early and also the equivalent `\parskip\z@skip` from my initial proposal.

The newer proposal makes sure `\parskip` is zero _after_  the `\rule` :  in this way the paragraph with the text of the annotation has a distance to the rule above it which is not influenced by the prevailing value of `\parskip` in the document. Similarly we want the bottom rule to be at a distance from the last line of the last paragraph of the annotation with no influence of a non-zero `\parskip`.  The new code above for `\py@endlightbox` has the same effect as my original proposal, it is only written differently.

**Comment by tk0miya** [comment]
New one looks more better.
I feel it is narrow the distance of the admonitions and its above elements.
New one resolved it.

I introduce the macro at 119790c.
It will released as 1.3.4 soon.

Thanks for your contribution!

**Comment by jfbu** [comment]
Yes I think you are right about former code narrowing distance with above elements.

Default value of `\parskip` in LaTeX is `0.0pt plus 1.0pt`. The former code set it to zero before drawing the rule on top of admonition, thus canceled the `plus 1.0pt` stretch component which TeX can use according to its criteria of what's beautiful on the page. Newer code sets its to zero only _after_ the rule (with the aim of having exactly one `\baselineskip` distance from the rule to the first line of the admonition).

Thanks for the commit !
